doc/devel/coding_style_guideline.rst

```diff
@@ -49,6 +49,182 @@ No alias should be used for `h5py`::
 
   import h5py
 
+-------------------
+Cython coding style
+-------------------
+DIPY recommends the use of the standard Python
+`PEP8 <https://www.python.org/dev/peps/pep-0008/>`_ style when writing
+`Cython <https://cython.org/>` code.
+
+Cython-specific syntax should follow these additional rules:
+
+Imports
+-------
+The `cimport`s should add the `c` prefix to the usual Python import package
+shorthand, e.g.::
+
+  cimport numpy as cnp
+
+Adding the `c` prefix to the import line makes it clear that the Cython/C
+symbols are being referred to as to compared to the Python symbols.
+
+Variable declaration
+--------------------
+Separate `cdef`, `cpdef`, and `ctypedef` statements from the following type by
+exactly one space. In turn, separate the type from the variable name by exactly
+one space. Declare only one `ctypedef` variable per line. You may `cdef` or
+`cpdef` multiple variables per line as long as these are simple declarations;
+note that multiple assignment, references, or pointers are not allowed on the
+same line. Grouping `cdef` statements is allowed. For example::
+
+  # Good
+  cdef int n
+  cdef char* s
+  cdef double Xf[3]
+  cdef double d[3]
+  cpdef int i, j, k
+  cdef Material mat = Material()
+  cdef:
+      double *ps = <double *> cnp.PyArray_DATA(seed)
+      cnp.npy_intp *pstr = <cnp.npy_intp *> qa.strides
+      cnp.npy_intp d, i, j, cnt
+      double direction[3]
+      double tmp, ftmp
+  cdef int true_enough(x):
+      return 1
+
+  # Bad
+  cdef  char *s
+  cdef char * s, * t, * u, * v
+  cdef double Xf[3], d[3]
+  cdef double x=42, y=x+1, z=x*y
+  cdef Material     mat   = Material()
+  cdef   int   falsified(x):
+      return 0
+
+Inside of a function, place all `cdef` statements at the top of the function
+body::
+
+  # Good
+  cdef int true_enough(x):
+      cdef int i = x
+      cdef int rtn
+      rtn = i + 42
+      return rtn
+
+  # Bad
+  cdef int falsified(x):
+      cdef int i = x, j = -42
+      j += i
+      cdef int rtn = j / j - 1
+      return rtn
+
+Using C libraries
+-----------------
+The `cimport`s should follow the same rules defined in PEP8 for `import`
+statements. If a module is both *imported* and *cimported*, the `cimport`
+should come before the `import`.
+
+An example of an imported C library::
+
+  from libc.stdlib cimport calloc, realloc, free
+
+Do not use `include` statements.
+
+Error return values
+-------------------
+When declaring an error return value with the `except` keyword, use one space on
+both sides of the `except`. If in a function definition, there should be no
+spaces between the error return value and the colon `:`. Avoid `except *`
+unless it is needed for functions returning `void`::
```

Comment:
There are a few places where

```
Avoid `except *` unless it is needed for functions returning `void`:
```
sentence is not fulfilled in DIPY, e.g. in `featurespeed.pyx:53`:
```
cdef void c_extract(Feature self, Data2D datum, Data2D out) nogil except *:
```

Reply:
I need to look deeper into this rule...